components/user_settings/display/user_settings_display.tsx

```diff
@@ -92,7 +92,7 @@ type Props = {
     configTeammateNameDisplay: string;
     currentUserTimezone: string;
     enableTimezone: boolean;
-    shouldAutoUpdateTimezone: boolean;
+    shouldAutoUpdateTimezone: boolean | string;
```

Comment:
When would this be a `string`? Sounds like we might need to convert a parameter somewhere.

Reply:
Is a typescript type error. I change this because in the mapStateToProps for UserSettingsDisplay, with makeGetUserTimezone we return a UserTimezone type now and the type of the property is boolean | string.
![image](https://user-images.githubusercontent.com/103535117/167144321-db45e708-955e-41b3-bdc2-3ada4bb6a222.png)
![image](https://user-images.githubusercontent.com/103535117/167145091-81d3e2a1-e924-4323-badb-3e7cd8ea148a.png)
Before this we would return only an blank object so there was no warning.

Reply:
If we change the type of `useAutomaticTimezone` to remove the `string`, what errors do we get?

Just wondering if there's an opportunity to clean up the types a bit here.

Reply:
I'm on going to remove the `string` type. 
I thought it would be simple. Unit tests pass but some e2e tests failed after removing and updating to boolean.
I will look further.

Reply:
After further investigation, the `string` type can't be remove.
`useAutomaticTimezone` is a string type server side in the api, so UserTimezone is return as a string during mapStateToProps. 
![image](https://user-images.githubusercontent.com/103535117/167203486-91c71b36-7e9b-4d22-8dc0-5bf38bf83062.png)
That's why both types are there.